src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1159,7 +1161,7 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
                             if (connector.CommandCts.IsCancellationRequested)
                             {
                                 connector.CommandCts.Dispose();
-                                connector.CommandCts = new CancellationTokenSource();
+                                connector.CommandCts = new TimeoutCancellationTokenSourceWrapper();
```

Comment:
This is a bit smelly to me - the whole point of this wrapper is to encapsulate all the checks and reallocation within it. A Restart method could help (the Stopwatch API is a good inspiration source here).

Reply:
Yeah, I thought about a method that resets the wrapper to definitely contain an uncancelled `CancellationTokenSource` but without actually starting the timeout.
Like calling `Start()` and then immediately calling `Stop()`.

I'll add that and change the code here.

Reply:
Yeah. Maybe Reset or Restart (though see discussion below about Restart)